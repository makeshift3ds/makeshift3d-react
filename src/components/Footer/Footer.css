@import '../../assets/css/globals.css';

footer {
  padding: 1rem;
  background-color: var(--brown);
  color: white;
  text-align: center;
}