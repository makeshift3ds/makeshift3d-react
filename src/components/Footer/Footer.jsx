import React from 'react';
import { getYear } from 'date-fns';

let currentYear = getYear(Date.now());

const Footer = () => (
  <footer>
    <div className="copyright">Copyright &copy;{currentYear}</div>
  </footer>
);

export default Footer;
