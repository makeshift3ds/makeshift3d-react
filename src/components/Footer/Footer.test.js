import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer.jsx';

describe('<Footer />', () => {
  it('should render 1 <Footer /> component', () => {
    const component = shallow(<Footer />);
    expect(component).toHaveLength(1);
  });

  it('should render an copyright element', () => {
    const component = shallow(<Footer />);
    expect(component.find('.copyright')).toHaveLength(1);
  });

  it('should match snapshot', () => {
    const component = shallow(<Footer />);
    expect(component).toMatchSnapshot();
  });
});
