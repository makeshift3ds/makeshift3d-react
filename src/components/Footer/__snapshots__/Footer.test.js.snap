// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="copyright"
      >
        Copyright ©
        2018
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Copyright ©",
          2018,
        ],
        "className": "copyright",
      },
      "ref": null,
      "rendered": Array [
        "Copyright ©",
        2018,
      ],
      "type": "div",
    },
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="copyright"
        >
          Copyright ©
          2018
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Copyright ©",
            2018,
          ],
          "className": "copyright",
        },
        "ref": null,
        "rendered": Array [
          "Copyright ©",
          2018,
        ],
        "type": "div",
      },
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
